### PR TITLE
updates Keras and tensorflow version [MNT]

### PR DESCRIPTION
**Pull request by solegalli**
At the moment, there are 4 tests that are failing in `test_keras.py`. But more important than that, Keras has migrated its `backend` functionality.

Currently, we do `import keras.backend as K` and from K we use functions like `argmax`, `gradient` and so on. Those functions don't live there any more. I think, we may find them in `tensforflow.keras.backend`, but I am not absolutely sure. Anyhow, if we swap `keras.backend` for `tensorflow.keras.backend`, then we get other errors, like keras can't accept a tensorflow tensor.

I am not an avid Keras/tensorflow user, so I am a bit lost here. Any help/pointers will be welcome!

Anyhow, i think we need to update the files explain_prediction.py and grandcam.py inside the keras folder inside eli5.

FYI @lopuhin @kmike

### CODE REVIEW

**Comment by lopuhin** [comment]
@solegalli I think with current state of Keras 3, it seems better to target using `keras` instead of `tensorflow.keras`, even if we still will only support/test tensorflow backend of keras. Some operations like argmax are present in `keras.ops`: https://keras.io/api/ops/numpy/#argmax-function but porting the rest of the code could be non-trivial.

**Comment by lopuhin** [comment]
See also https://github.com/eli5-org/eli5/pull/31 which seems to add support for more recent TF versions

**Comment by lopuhin** [comment]
Thanks for starting this @solegalli , let me close this one as now we support modern Keras via #58
